ekncontent/ekncontent/eknc-query-object.c

```diff
@@ -879,12 +879,18 @@ static XapianQuery *
 get_content_type_clause (const char *content_type)
 {
   g_autofree char *prefixed = NULL;
+  XapianQuery *retval = NULL;
 
   if (content_type == NULL)
     return NULL;
 
   prefixed = g_strconcat (XAPIAN_PREFIX_CONTENT_TYPE, content_type, NULL);
-  return xapian_query_new_for_term (prefixed);
+  if (g_str_has_suffix (content_type, "*"))
+      retval = xapian_query_new_wildcard (g_strstrip (g_strdelimit (prefixed, "*", ' ')));
```

Comment:
This will convert all `*` in the string into `' '`. Probably an edge case to get something like `foo*/bar*`, but perhaps it'd be better to explicitly chop characters off the end using a separate helper function.

Reply:
IIRC Xapian only supports trailing wildcards, not possible to do something like `foo*/bar*`.